### PR TITLE
Addition of scoreboard event

### PR DESCRIPTION
**Pull request by Ascynx**
a few changes (also for some reason last merge didn't count on here and so the old pull request commits are here too :skull:)

### CODE REVIEW

File: src/main/java/mod/gate/Gate.java [hunk 1]
```diff
@@ -1,17 +1,13 @@
 package mod.gate;
 
 import com.google.gson.JsonParseException;
-import com.mojang.blaze3d.systems.RenderSystem;
-import mod.gate.config.Configuration;
-import mod.gate.events.EventHandler;
+import mod.gate.core.config.Configuration;
+import mod.gate.core.events.EventHandler;
+import mod.gate.utils.*;
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/Gate.java]: No wild card imports please

**Comment by Ascynx** [comment]
actual changes start after [this commit](https://github.com/Cobble8/Gate/pull/5/commits/667719969a32bee85318abc370eb0946fd01a81f)

File: src/main/java/mod/gate/core/events/ingamehud/ActionbarReceiveEvent.java [hunk 2]
```diff
@@ -1,4 +1,4 @@
-package mod.gate.events;
+package mod.gate.core.events.ingamehud;
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/events/ingamehud/ActionbarReceiveEvent.java]: Why are you making this a package? That's no, please don't do this

File: src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(ClientPlayNetworkHandler.class)
+public class ScoreboardPacketMixin {
+    private long lastUpdateAt = 0;
+
+    @Inject(at= @At("TAIL"), method = "onTeam")
+    public void onTeamMixin(TeamS2CPacket packet, CallbackInfo ci) {
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java]: Stop using autocomplete, call ci -> info

File: src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+@Mixin(ClientPlayNetworkHandler.class)
+public class ScoreboardPacketMixin {
+    private long lastUpdateAt = 0;
+
+    @Inject(at= @At("TAIL"), method = "onTeam")
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java]: Why do you inject at tail?

File: src/main/java/mod/gate/utils/Scoreboard.java [hunk 5]
```diff
@@ -25,9 +26,16 @@ public static String getCurrentSubArea() {
...
+    private static ArrayList<String> Lines = new ArrayList<>();
+    public static ArrayList<String> getLines() {
+        return Lines;
+    }
+
 
     //TODO Add event to run this command
-    private static void loadScoreboard() {
+    public static void loadScoreboard() {
+        System.out.println("triggered loadScoreboard " + System.currentTimeMillis());
```
Comment (Ascynx) [inline src/main/java/mod/gate/utils/Scoreboard.java]: noticed that isn't supposed to be there, It'll be removed in next commit

File: src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    public void onTeamMixin(TeamS2CPacket packet, CallbackInfo ci) {
+        //trigger post operations
+
+        if (lastUpdateAt + 10000 < System.currentTimeMillis()) {//limits the update to every 10 seconds
```
Comment (DalwynWasTaken) [inline src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java]: Why do we want this limit? Don't we want accurate info? We can change internal scoreboard class to improve speed but this shouldn't be a solution if it's laggy

**Review by DalwynWasTaken — COMMENTED** [review]
I personally think this entire event should be scraped and we should just call loadScoreboard on a getter of the info in Scoreboard class since it doesn't need event data

File: src/main/java/mod/gate/Gate.java (hunk 1 above)
Comment (Ascynx) [reply src/main/java/mod/gate/Gate.java]: didn't even know I added a wildcard import, so my guess is intelliJ did it

File: src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java (hunk 4 above)
Comment (Ascynx) [reply src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java]: was to do it once every change of scoreboard was done

File: src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java (hunk 6 above)
Comment (Ascynx) [reply src/main/java/mod/gate/core/mixin/ScoreboardPacketMixin.java]: wasn't really laggy from what I noticed but thought it wasn't necessary to update more than once every 10 seconds

**Review by DalwynWasTaken — APPROVED** [review]

**Pull request merged**